mlir/lib/Gradient/IR/GradientOps.cpp

```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <assert.h>
```

Comment:
Inconsequential, but since Lee wants us to use C++ over C as much as possible:
```suggestion
#include <cassert>
```